Use Amazon's Shopping Queries Dataset (#26)
chlorochrule/solrcloud-development-box: README.md

Development environment for SolrCloud using Docker and Docker compose.
This repository provides customized build for any lucene commit and solr commit.

> ## How to use: create solr system

Reviewer: ```suggestion
## How to use: Setup SolrCloud
```

### prepare json data

Amazon Shopping Queries Dataset is able to be got from [here](https://github.com/amazon-science/esci-data).

This dataset is included as a submodule in this repository.

You can prepare json data by the following steps:

1. take the data from github, this step may take a long time

    ```bash
    git submodule update --init --recursive
    ```

2. generate dataset
    ```bash
    cd data
    python3 create_shopping_data.py
    cd ..
    ```

then you can check you json data in folder data/amazonshopping

> ### launch dolr cloud

Reviewer: typo
```suggestion
### launch SolrCloud
```

Run:

```bash
docker-compose up -d --build
```

You can see the solr system by accessing `localhost:8983` or `localhost:8984`.

> The pull down menu 「Collection Selector」is empty now.

Reviewer: use ascii
```suggestion
The pull down menu "Collection Selector" is empty now.
```

### Upload configsets and create collection

Configsets of amazon shopping data collection can be uploaded by:

```bash
# upload configsets
docker-compose exec solr1 /opt/solr/server/scripts/cloud-scripts/zkcli.sh -zkhost zoo:2181 -cmd upconfig -confdir /opt/solr/configsets/amazonshopping -confname amazonshopping
# create collection
curl 'localhost:8983/solr/admin/collections?action=CREATE&name=amazonshopping&numShards=2&replicationFactor=2&maxShardsPerNode=2&collection.configName=amazonshopping'
```

The file `solr/configsets/amazonshopping` is created base to the json file in `data/amazonshopping/**.json`,

The `field` defines every fields in json file, with the primary key named `example_id`.

> Now, access to `localohst:8983`, you can find the `amazonshopping` included in the pull down menu 「Collection Selector」.

Reviewer: ```suggestion
Now, access to `localohst:8983`, you can find the `amazonshopping` included in the pull down menu "Collection Selector".
```

> ※ Here, you can find solr API [here](https://solr.apache.org/guide/7_1/coreadmin-api.html#coreadmin-create).

Reviewer: ```suggestion
You can find Solr API [here](https://solr.apache.org/guide/solr/latest/configuration-guide/coreadmin-api.html#coreadmin-create)
```

### Feed documents

> The json file prepared before can be fee by curl command:

Reviewer: typo
```suggestion
The json file prepared before can be feed by curl command:
```
